src/new_skills/NewSkill.cpp

```diff
@@ -0,0 +1,136 @@
+//
+// Created by jessevw on 21.02.20.
+//
+
+#include "include/roboteam_ai/new_skills/NewSkill.h"
+#include "skills/Skill.h"
+
+#include <include/roboteam_ai/utilities/Settings.h>
+
+#include <cmath>
+
+#include "control/ControlUtils.h"
+#include "utilities/Constants.h"
+#include "utilities/GameStateManager.hpp"
+#include "utilities/RobotDealer.h"
+#include "world/Ball.h"
+#include "world/Robot.h"
+#include "world/World.h"
+
+namespace rtt::ai {
+
+    NewSkill::NewSkill(std::string name, bt::Blackboard::Ptr blackboard) : bt::Leaf(std::move(name), std::move(blackboard)) {
+        robot = std::make_shared<Robot>();
+        ball = std::make_shared<Ball>();
```

Comment:
I guess this was taken from the old skil. Yet, if you assign the robot and the ball at initialize anyway, there is not really a need to create them here.

Reply:
Yeah you are right I think